### PR TITLE
Feature ordering missions api

### PR DESCRIPTION
**Pull request by GodVenn**
This is important for pagination to make sense, you can't order on the client side if the response is paginated 
You can order the 10 items to be ordered by end time for example, but that page might still not hold the most recent end times
You need to be able to order on server side for this, and that's what this pull request does

### CODE REVIEW

**Comment by GodVenn** [comment]
> 🔔 Changes in database folder detected 🔔 Do these changes require **adding new migrations**? 🤔 In that case follow [these steps](https://github.com/equinor/flotilla/tree/main/backend#Database-model-and-EF-Core). If you are uncertain, ask a database admin on the team 😄

Not needed

File: backend/api/Services/MissionService.cs [hunk 1]
```diff
@@ -346,5 +354,59 @@ MissionQueryStringParameters parameters
...
+                if (objectProperty == null)
+                    throw new InvalidDataException(
+                        $"Mission has no property '{propertyFromQueryName}' for ordering"
+                    );
+                string sortingOrder = param.EndsWith(" desc", StringComparison.OrdinalIgnoreCase)
+                  ? "descending"
+                  : "ascending";
+#pragma warning disable CA1305
+                orderQueryBuilder.Append($"{objectProperty.Name} {sortingOrder}, ");
+#pragma warning restore CA1305
```
Comment (tsundvoll) [inline backend/api/Services/MissionService.cs:399]: Why is this needed?

File: backend/api/Services/MissionService.cs [hunk 2]
```diff
@@ -346,5 +354,59 @@ MissionQueryStringParameters parameters
...
+            }
+
+            string orderQuery = orderQueryBuilder.ToString().TrimEnd(',', ' ');
+            if (string.IsNullOrWhiteSpace(orderQuery))
+            {
+                missions = missions.OrderBy(x => x.Name);
```
Comment (tsundvoll) [inline backend/api/Services/MissionService.cs:405]: Can we choose a more descriptive name than `x`?

File: backend/api/Database/Context/FlotillaDbContext.cs [hunk 3]
```diff
@@ -12,25 +14,88 @@ public class FlotillaDbContext : DbContext
...
+                            pose =>
+                            {
+                                pose.OwnsOne(p => p.Position);
+                                pose.OwnsOne(p => p.Orientation);
+                            }
+                        );
+                    }
+                );
+            }
+        );
```
Comment (tsundvoll) [inline backend/api/Database/Context/FlotillaDbContext.cs:52]: Can we have more descriptive variable names than `m`, `t`, `r` and `p`?

File: backend/api/Services/MissionService.cs (hunk 2 above)
Comment (GodVenn) [reply backend/api/Services/MissionService.cs:405]: Yes!

**Review by tsundvoll — APPROVED** [review]
Fix comments above, then it looks good to me :+1:
